queue/driver_mongo.go

```diff
@@ -410,7 +456,16 @@ func (d *mongoDriver) modifyQueryForGroup(q bson.M) {
 func (d *mongoDriver) Get(ctx context.Context, name string) (amboy.Job, error) {
 	j := &registry.JobInterchange{}
 
-	res := d.getCollection().FindOne(ctx, bson.M{"_id": d.getIDFromName(name)})
+	matchID := bson.M{
+		"$or": []bson.M{
+			{"_id": d.getIDWithGroup(name)},
+			{"retry_info.base_job_id": name},
```

Comment:
It's not supposed to have the group name because users aren't supposed to be aware of the format of the stored `_id`, only the logical job ID. I purposely wrote `addMetadata` and `removeMetadata` so that when jobs go into the DB, the base job ID is set before changing the `_id`.

Putting the same job in two different group queues would be invalid. I don't see a good use case for putting the same job in two logically separate queues (and typically it just leads to bugs), because it'll just cause your job to run twice instead of once.

Reply:
I think my confusion is that this is an `$or`, so it seems like group name for retryable jobs is pointless? I guess I am confused on the point of prepending the group name then at all.

Reply:
The group name is not used in base job ID. The group name prepending is the current behavior - it has to be prepended for backward compatibility.